feat(kdatetimepicker): port to beta branch (#803)
Kong/kongponents: docs/components/datetime-picker.md

  mode="relative"
  width="480"
  :minDate="minDate"
  :maxDate="maxDate"
  :range="true"
  :timePeriods=[
    {
      section: "Last",
      values: [
        { key: "15m", prefix: "Last", timeframeText: "15 minutes", start: function() {}, end: function() {} },
        { key: "12h", prefix: "Last", timeframeText: "12 hours", start: function() {}, end: function() {} },
        { key: "24h", prefix: "Last", timeframeText: "24 hours", start: function() {}, end: function() {} },
        ...
      ]
    },
    {
      section: "Current",
      values: [
        { key: "current_week", prefix: "Current", timeframeText: "week", start: function() {}, end: function() {} },
        { key: "current_month", prefix: "Current", timeframeText: "month", start: function() {}, end: function() {} }
      ]
    }
    {
      section: "Previous",
      values: [
        { key: "previous_week", prefix: "Previous", timeframeText: "week", start: function() {}, end: function() {} },
        { key: "previous_month", prefix: "Previous", timeframeText: "month", start: function() {}, end: function() {} }
      ]
    }
  ]
/>
```

## Props

### v-model

#### Single date time picker `v-model`

> A single date or time picker instance which can be seeded with a value like so:
> 
> ```js
> <KDateTimePicker
>   v-model="currentValue"
>   ...
> />
> Where `currentValue` is a valid Date object, eg: `new Date()`
> ```

Reviewer: The formatting here is incorrect.

The code block should be of type `html` and then the "Where..." text just regular text. Something like this might be easier to read IMO.

Same comment applies below for a lot of the props sections.

```suggestion
A single date or time picker instance which can be seeded with a value as shown here, where `currentValue` is a valid Date object, eg: `new Date()`.

```html
<KDateTimePicker
  v-model="currentValue"
/>
```
```

#### Range date time picker `v-model`

An instance that displays a date range or a series of relative time frames.

> Can be seeded with a **relative time frame** by setting `v-model` to:

Reviewer: ```suggestion
Can be seeded with a **relative time frame** by setting the `v-model` to `currentValue` as shown here:
```

```js
currentValue = {
  start: '',
  end: '',
  timePeriodsKey: TimeframeKeys.ONE_DAY
},
```

Or it can be seeded with a **custom calendar range** by setting `v-model` to:

> ```js

Reviewer: Utilize `ts` blocks for the script code instead of js

```suggestion
```ts
```

**default**: `null`

### maxDate

A valid `Date` object (eg: `Fri Aug 19 2022 12:00:00 GMT-0700 (Pacific Daylight Time)`) which blocks out days/times that occur **after** the given timestamp. `maxDate` gets passed down to the calendar component, and does not apply to relative time frames.

**default**: `null`

### minuteIncrement

A `Number` which sets the custom interval for the Minutes select dropdown.

**default**: `5`

### mode

Required `String` prop which accepts the following values: `date`, `time`, `dateTime` and `relative`. The `relative` value is meant to denote an datetime picker instance which only contains relative time frames.

### placeholder

`String` containing the default message displayed in the date time picker input. Clearing any selected values will revert the input to display this placeholder message.

**default**: `Select a time range`

### range

`Boolean` which determines whether the calendar allows selection of a **single** date or time, as opposed to a **range** of start and end values.

**default**: `false`

### timePeriods

> An array of time frame values to be displayed as buttons in the "Relative" section of the popover.  Please note that each timeframe's `key` attribute needs to be unique.

Reviewer: Let's emphasize that the `key` needs to be unique.

```suggestion
An array of time frame values to be displayed as buttons in the "Relative" section of the popover.  

**Please note that each timeframe's `key` attribute must be unique.**
```

**default**: `[]`

**Example:**

```js
[
  {
    section: "Last",
    values: [
      { key: "15m", prefix: "Last", timeframeText: "15 minutes", start: () => {}, end: () => {} },
      { key: "12h", prefix: "Last", timeframeText: "12 hours", start: () => {}, end: () => {} },
      { key: "24h", prefix: "Last", timeframeText: "24 hours", start: () => {}, end: () => {} },
    ]
  },
   {
    section: "Current",
    values: [
      { key: "current_week", prefix: "Current", timeframeText: "week", start: () => {}, end: () => {} },
      { key: "current_month", prefix: "Current", timeframeText: "month", start: () => {}, end: () => {} },
    ]
  }
  {
    section: "Previous",
    values: [
      { key: "previous_week", prefix: "Previous", timeframeText: "week", start: () => {}, end: () => {} },
      { key: "previous_month", prefix: "Previous", timeframeText: "month", start: () => {}, end: () => {} },
    ]
  }
]
```

## Events

The events below will fire whenever **Apply** is clicked (assuming the user has made a custom calendar or relative timeframe selection). If **Clear** is clicked, the object shape remains the same, but the values are empty strings.

### Single date time picker

> **Event**: `input` and `change`

Reviewer: ```suggestion
**Events**: `input`, `change`, `update:modelValue`
```

Will emit a JavaScript Date object, or an empty string  "Clear".

```js
"2022-09-06T19:45:54.406Z"  // ISO 8601 Date string
```

### Range date time picker

> **Event**: `input` and `change`

Reviewer: ```suggestion
**Events**: `input`, `change`, `update:modelValue`
```
